tmail-backend/tmail-third-party/openpaas/src/main/java/com/linagora/tmail/CalDavEventAttendanceRepository.java

```diff
@@ -0,0 +1,286 @@
+            AttendanceStatus.fromCalendarAttendeeParticipationStatus(attendanceStatus).orElseThrow());
+    }
+
+    private Mono<DavUser> toDavUser(Username username) {
+        try {
+            return davUserProvider.provide(username.asMailAddress())
+                .switchIfEmpty(Mono.error(new RuntimeException("Unable to find user in Dav server with username '%s'".formatted(username.asString()))));
+        } catch (AddressException e) {
+            LOGGER.debug("Failed to get user DAV information", e);
+            return Mono.empty();
+        }
+    }
+
+    @Override
+    public Publisher<CalendarEventReplyResults> setAttendanceStatus(Username username,
+                                                                    AttendanceStatus attendanceStatus,
+                                                                    BlobIds eventBlobIds,
+                                                                    Optional<LanguageLocation> maybePreferredLanguage) {
+        MailboxSession session = sessionProvider.createSystemSession(username);
+
+        return toDavUser(username)
+            .flatMapMany(davUser ->
+                blobIdsAsFlux(eventBlobIds)
+                    .flatMap(blobId -> updateAttendanceStatusForEventBlob(blobIdFromRefinedId(blobId), attendanceStatus, davUser, session)));
+    }
+
+    private Mono<CalendarEventReplyResults> updateAttendanceStatusForEventBlob(BlobId blobId, AttendanceStatus attendanceStatus, DavUser davUser, MailboxSession session) {
+        PartStat partStat = attendanceStatus.toPartStat().orElseThrow();
+        return extractMessageId(blobIdToString(blobId))
+            .flatMap(messageId ->
+                Mono.fromDirect(messageIdManager.getMessagesReactive(List.of(messageId), FetchGroup.HEADERS, session))
+                    .map(this::retrieveEventUid)
+                    .flatMap(eventUid -> doGetCalendarObjectContainingVEvent(davUser, eventUid))
+                    .flatMap(targetCalendarObject ->
+                        updateCalendarObject(davUser, targetCalendarObject.uri(), partStat))
+                    .thenReturn(CalendarEventReplyResults$.MODULE$.done(blobId))
+                    .onErrorResume(e -> Mono.just(CalendarEventReplyResults$.MODULE$.notDone(blobId, e, session))));
+    }
+
+    private Mono<Void> updateCalendarObject(DavUser davUser, URI calendarObjectURI, PartStat partStat) {
+        return davClient.updateCalendarObject(davUser, calendarObjectURI,
+            (calendarObject -> createUpdatedCalendar(calendarObject, davUser.username(), partStat)));
+    }
+
+    private Mono<DavCalendarObject> doGetCalendarObjectContainingVEvent(DavUser davUser, String eventUid) {
+        return davClient.getCalendarObjectContainingVEvent(davUser, eventUid)
+            .switchIfEmpty(
+                Mono.error(
+                    new RuntimeException("Unable to find any calendar objects containing VEVENT with id '%s'".formatted(eventUid))));
```

Comment:
```suggestion
            .switchIfEmpty(Mono.error(() -> new RuntimeException("Unable to find any calendar objects containingVEVENT with id '%s'".formatted(eventUid))));
```

Appart from the ident, please don't instanciate a stacktrace eagerly when no exception needs to be thrown...